### ISSUE TITLE
Content for a Wiki

### ISSUE THREAD

**Issue by Bobbicat** [jimbob88/wheelers-wort-works#15]
There is a stack of information, that will be needed by those who come along to look at this application.
I imagine it will continue across platform and be available via browser and mobile device too. As or when these things happen brewers will need to know. Links to this point can provide information.

 We can very quickly get ourselves bogged down with complication, so let us begin.

Lets keep it simple for a start. 
Lets look at who our audience will be and make sure we are addressing them and that they understand us.
Let us look at how Graham would have liked this to go and decide if that is the way forward.
For a start lets tell what is offered, who can use this, and be prepared for feedback, in fact it is important to provide a method for feedback.
That is for starters.

---

Here is some content as an introduction, feel free to use, to reject or to modify it:

---

WIKI

A short introduction to WWW

HISTORY
Some years ago Graham Wheeler, a well respected author, created an application called Beer Engine. It was an uncomplicated easy to use tool, that produced good results, intended to aid home brewers.
He deliberately avoided adding extra bells and whistles. His aim was that it should work well, be useful and be available to both beginners and old hands.

Sadly, Graham is no longer with us. It is hoped that Wheelers Wort Works will continue his legacy and provide an application that is as close to his original intention as it is possible to get.

---
This application can be used to plan, inspect and/or edit a brew. It will predict how a brew will turn out when the ingredients are entered into the program and will show how the outcome will change if the contents are modified.
There are options to edit the grist, hop, and yeast databases, so extra components can be entered and their contents changed as and when needed.

There will be a set of simple walk through instructions to demonstrate the use of each part of Wheelers Wort Works.

The first section will cover how to obtain a copy of the application, how to set it up and how to update.

---

**Comment by jimbob88** [comment]
Thank you very much, I have made amendments and this is now the front page of the Wiki :+1:

**Comment by Bobbicat** [comment]
If you feel the need to edit or make changes to anything I put up here I am okay with that.
I'm sure you will have your own take on how you want this to appear.
This is your project after all.

**Comment by Bobbicat** [comment]
How about somethig like this:

Wheelers Wort Works is referred to as 'WWW' in this document for the sake of brevity.

1. Getting Wheelers Wort Works onto your device. (You would have to create this section, I don't have the specifics for operating systems different from my own.)
    WWW is an on going project so you will need to update regularly to get the best from it
    Outline of update methods for each OS.
(I know this is a repeat of stuff you have put up elsewhere but I think your Wiki should be seen as a centre for all relevant info and be where a user or prospective user would go to get instruction and info
and the first thing will be why to get it and how to get it and who it is aimed at)

2.  Using WWW for the first time.
 This application and its Wiki is not intended to be an all-inclusive guide to brewing, but WWW is aimed at beginners and old hands alike.
 To begin you can discover how to brew on line, join a Forum, join a Club or buy a book (or two).
 You will need a some equipment, some ingredients and a recipe.
 This Wiki assumes that you have already made a start with home brewing, though you might be a 
 beginner.

 a) A short demo/walkthrough - setting up a simple all grain brew in WWW.
      i) The Engine Room
     ii) The first recipe
     iii) Simple edits in the Engine Room

 b) Why alter the Grist and Hop data? 
     i) Adding new ingredients. 
     ii) How to edit Grist and Hop.

 c) Editing Defaults. Why and How. 

 x) Setting up a partial mash (what is a partial mash?) [I don't do this so it will need extra input, I'm not even sure if partial mash brews will work correctly here.]

All the above will need further editing and expanding upon, it isn't ready for a direct cut and paste.

**Comment by jimbob88** [comment]
I will be working a lot on the Wiki across the next few days. It'll be an interesting thing to do in my free time and I am sure I can make a lot of progress into all these outlined ideas.
Thank you so much for the ideas of how to begin work on this project.

**Comment by Bobbicat** [comment]
There is a lot to work through there but if you go at it steadily a bit at a time you will get there. Making WWW accessible for the complete novice will get a large number of followers, but I'm sure old hands will give it a try too, if they realise its potential.
I see you have made good progress with the wiki. It is coming along well.

**Comment by Bobbicat** [comment]
I see you have been making more progress. That is great!
Here's a little more from me. Again, edit, use as you wish, of course:

---

Advanced features in the Engine Room:

**Quit button**

On pressing gives option to OK or cancel,  then, if OK, option: save or cancel[cancel=quit + no save]

**Remove button**

If you wish to remove, for whatever reason, an item from the Fermentable table, select it and click the Remove button just under the table.
There is a similar button under the Hop table. 

**Zero buttons**

There is a Zero button next to the Fermentable table and another next to the Hop table.
If you select an ingredient in one of the tables and press the appropriate Zero button the quantities of that ingredient will be reset to zero. 

**Alpha buttons**

Very probably an actual Hop you own and wish to use has a different Alpha value from the default shown when you choose it in the Engine Room. 
For example you might have some Golding with an actual Alpha described on its packaging as 4.2. This will affect the bitterness produced by your hop.
To compensate, first add the hop and weight to the hop table. Then make a note of the displayed IBU result for Golding [for example 12]. Select Golding which has a default Alpha value of 5.7 and use the Alpha buttons to shift Alpha to the value for your hop [4.2].
This will affect the IBU reading now given for Golding. Change the weight of Golding until the IBU for Golding is back at the figure you originally noted [in this case 12].
The effective bittering with your hop will now be identical to the original recipe. 

**Time buttons**

By default the Hop window displays a time of 60 minutes. [usually the time for a complete boil]
You might want to change the time to match your recipe.[these days there are such things as thirty and forty-five minute boils, for example]
As well as  having a 90 minute boil time, sometimes a hop is added 'late', perhaps for the last 10-15 minutes of the boil.
Select the particular hop you wish to adjust and use the time buttons to make the changes you require.

**Recalculate button**

[I am not sure I fully understand how and when this functions. Perhaps you will describe this, Jim?]

**Comment by Bobbicat** [comment]
A description of the following would also be useful.
What they show, how and why, to enter preset values[?], how, why, to use them.

**Recipe Name**
**Volume**
**Boil Volume**
**Original Gravity**
**Bitterness IBU**
[also the little tick boxes]

I only have a partial understanding of these bits of WWW

**Comment by Bobbicat** [comment]
I did see an instruction set for beer engine by Graham, on JBK forum I think. I'll see if I can find it again.

It is here at JimsBeerKit Forum:

[https://www.jimsbeerkit.co.uk/forum/viewtopic.php?f=77&t=17405#p202792](url)

towards the bottom of the page, 'Getting Started'.

**Comment by jimbob88** [comment]
Thank you very much, I'll start converting this over soon. Thanks for all your help :)

**Comment by jimbob88** [comment]
> 
> 
> A description of the following would also be useful.
> What they show, how and why, to enter preset values[?], how, why, to use them.
> 
> **Recipe Name**
> **Volume**
> **Boil Volume**
> **Original Gravity**
> **Bitterness IBU**
> [also the little tick boxes]
> 
> I only have a partial understanding of these bits of WWW

Yeah, I'll try and provide more light over what these are for and show.

**Comment by Bobbicat** [comment]
> Thank you very much, I'll start converting this over soon. Thanks for all your help :)

You are providing something that is useful to me and will also be good for others too, I've no doubt, so all thanks are due to you. It is a pleasure to contribute to this great application. Keep up the good work!

**Comment by Bobbicat** [comment]
I have taken the liberty of posting a link to your application on JBK Forums.

https://www.jimsbeerkit.co.uk/forum/viewtopic.php?f=77&t=17405&start=195

**Comment by jimbob88** [comment]
Hi again, it had been a while since I'd done much work on expanding the Wiki, but I am happy to announce the implementation of the `Advanced Series`. I have gone through and attempted to document as much of the Wheeler's UI as I could think of at the time. I am more than happy to explain different areas further and to expand them more as I understand for a while some sections of Wheeler's have been an oddity. 
Although I don't believe you should have to read the documentation for a program at every corner simply to be able to use it, I believe some of the more complex areas of Wheeler's Wort Works have become lost, perhaps due to a combination of odd naming schemes and also areas of code now directly linked to brewing!

Hopefully the new system will help when finding your way around the Wheeler's Wort Works User Interface. 

As another note I have been considering looking into adding what I am going to dub the `Help Update`, with the idea being a menu item you can press then your mouse cursor becomes a questionm-mark that allows you to query buttons and sections of the Wheeler's Wort Works Interface, with each button hopefully linking back to the GitHub Wiki for WWW. 

Best Regards,
Jim
